## Exercício 5 – Teste a hipótese de o salário ser o mesmo independente do grau de instrução. Qual sua conclusão?

In [1]:
%pip install statsmodels

Note: you may need to restart the kernel to use updated packages.


In [2]:
# importa biblioteca pandas
import pandas as pd

# importa biblioteca statsmodels
import statsmodels.api as sm

# local do arquivo
Caminho='/Users/fabicampanari/Desktop/_8-Prova Matematematica/1-statiscalMeasures_ Hypothesis Testing II./1🇧🇷-statiscalMeasures_ Hypothesis Testing II./answeredCodes_statiscalMeasures/cadastro_funcionarios.xlsx'


In [3]:
#Traz o arquivo para o Python
df=pd.read_excel(Caminho)


In [4]:
df.head( )


ID estado_civil grau_instrucao  n_filhos  salario  idade reg_proc
0   1            s              f         0     4.00     26      int
1   2            c              f         1     4.56     32      cap
2   3            c              f         2     5.25     36      cap
3   4            s              m         0     5.73     20      out
4   5            s              f         0     6.26     40      out

In [5]:
#Verifica a média de idade por grupo
df.groupby(['grau_instrucao'])['idade'].describe()


count       mean       std   min   25%   50%   75%   max
grau_instrucao                                                          
f               120.0  35.533333  6.925696  20.0  28.0  37.0  41.0  49.0
m               203.0  34.556650  6.145955  20.0  31.0  35.0  37.0  49.0
s               103.0  33.689320  6.159687  22.0  31.0  33.0  38.0  49.0

In [6]:
#Gera o modelo para comparar idade por grau de instrução
from statsmodels.formula.api import ols
modelo=ols('idade ~ grau_instrucao',data=df).fit()


In [7]:
#Aplicação ANOVA Testando H0: μf= μm=μs   H1: Ao menos um diferente
anova=sm.stats.anova_lm(modelo)
print(anova)


                   df        sum_sq    mean_sq         F    PR(>F)
grau_instrucao    2.0    190.129141  95.064570  2.336835  0.097879
Residual        423.0  17208.023441  40.680906       NaN       NaN


In [8]:
#Teste Post Hoc de Tukey para avaliar diferenças
from statsmodels.stats.multicomp import pairwise_tukeyhsd
tukey=pairwise_tukeyhsd(endog=df.idade, groups=df.grau_instrucao)
tukey.summary ( )


<class 'statsmodels.iolib.table.SimpleTable'>

In [9]:
#Gera o modelo para comparar idade por região de procedencia
from statsmodels.formula.api import ols
modelo=ols('idade ~ reg_proc',data=df).fit()
#Aplicação ANOVA Testando H0: μf= μm=μs   H1: Ao menos um diferente
anova=sm.stats.anova_lm(modelo)
print(anova)

             df        sum_sq     mean_sq         F    PR(>F)
reg_proc    2.0    454.970401  227.485201  5.679349  0.003682
Residual  423.0  16943.182181   40.054804       NaN       NaN


In [10]:
#Verifica a média de idade por grupo
df.groupby(['reg_proc'])['idade'].describe()


count      mean       std   min   25%   50%   75%   max
reg_proc                                                         
cap       143.0  36.06993  5.550854  22.0  33.0  35.0  39.5  49.0
int       135.0  33.77037  6.004899  20.0  29.5  33.0  37.5  49.0
out       148.0  34.00000  7.254368  20.0  28.0  34.0  40.0  46.0

In [11]:
#Teste Post Hoc de Tukey para avaliar diferenças
from statsmodels.stats.multicomp import pairwise_tukeyhsd
tukey=pairwise_tukeyhsd(endog=df.idade, groups=df.reg_proc)
tukey.summary ()

<class 'statsmodels.iolib.table.SimpleTable'>

In [12]:
#Verifica a média de idade por grupo
df.groupby(['grau_instrucao'])['salario'].describe()

count       mean       std   min      25%     50%     75%  \
grau_instrucao                                                              
f               120.0   7.922667  2.802770  4.00   6.2475   7.365   8.995   
m               203.0  11.208522  3.785457  4.43   8.2600  10.500  13.850   
s               103.0  14.507864  3.590439  7.60  11.3500  14.550  17.260   

                 max  
grau_instrucao        
f               14.7  
m               23.9  
s               23.5

In [13]:
#Gera o modelo para comparar idade por grau de instrução
from statsmodels.formula.api import ols
modelo=ols(' salario ~ grau_instrucao',data=df).fit()


In [14]:
#Gera o modelo para comparar idade por região de procedencia
from statsmodels.formula.api import ols
modelo=ols('salario ~ grau_instrucao',data=df).fit()
#Aplicação ANOVA Testando H0: μf= μm=μs   H1: Ao menos um diferente
anova=sm.stats.anova_lm(modelo)
print(anova)

                   df       sum_sq      mean_sq          F        PR(>F)
grau_instrucao    2.0  2409.879383  1204.939691  99.078295  5.116294e-36
Residual        423.0  5144.310233    12.161490        NaN           NaN


In [15]:
#Teste Post Hoc de Tukey para avaliar diferenças
%pip install statsmodels
from statsmodels.stats.multicomp import pairwise_tukeyhsd
tukey=pairwise_tukeyhsd(endog=df.salario, groups=df.grau_instrucao)
tukey.summary()

Note: you may need to restart the kernel to use updated packages.


<class 'statsmodels.iolib.table.SimpleTable'>

## Questao 5 -  Teste a hipótese de o salário ser o mesmo independente do grau de instrução. Qual sua conclusão?

In [16]:
# Importa bibliotecas necessárias
import pandas as pd
import statsmodels.api as sm
from statsmodels.formula.api import ols
from statsmodels.stats.multicomp import pairwise_tukeyhsd

In [17]:
# Carrega o arquivo Excel
caminho = '/Users/fabicampanari/Desktop/_8-Prova Matematematica/1-statiscalMeasures_ Hypothesis Testing II./1🇧🇷-statiscalMeasures_ Hypothesis Testing II./answeredCodes_statiscalMeasures/cadastro_funcionarios.xlsx'
df = pd.read_excel(caminho)


In [18]:
# Visualiza as primeiras linhas do DataFrame
print(df.head())

   ID estado_civil grau_instrucao  n_filhos  salario  idade reg_proc
0   1            s              f         0     4.00     26      int
1   2            c              f         1     4.56     32      cap
2   3            c              f         2     5.25     36      cap
3   4            s              m         0     5.73     20      out
4   5            s              f         0     6.26     40      out


In [19]:
# Verifica a média de salário por grupo de grau de instrução
print(df.groupby(['grau_instrucao'])['salario'].describe())

                count       mean       std   min      25%     50%     75%  \
grau_instrucao                                                              
f               120.0   7.922667  2.802770  4.00   6.2475   7.365   8.995   
m               203.0  11.208522  3.785457  4.43   8.2600  10.500  13.850   
s               103.0  14.507864  3.590439  7.60  11.3500  14.550  17.260   

                 max  
grau_instrucao        
f               14.7  
m               23.9  
s               23.5  


In [20]:
# Gera o modelo para comparar salário por grau de instrução
modelo = ols('salario ~ grau_instrucao', data=df).fit()
modelo

In [21]:
# Aplicação ANOVA
anova = sm.stats.anova_lm(modelo)
print("Resultados do Teste ANOVA:")
print(anova)

Resultados do Teste ANOVA:
                   df       sum_sq      mean_sq          F        PR(>F)
grau_instrucao    2.0  2409.879383  1204.939691  99.078295  5.116294e-36
Residual        423.0  5144.310233    12.161490        NaN           NaN


In [22]:
# Interpretação dos resultados
if anova['PR(>F)'][0] < 0.05:
    print("Conclusão: Existe uma diferença significativa nos salários entre os diferentes graus de instrução.")
else:
    print("Conclusão: Não existe uma diferença significativa nos salários entre os diferentes graus de instrução.")

Conclusão: Existe uma diferença significativa nos salários entre os diferentes graus de instrução.


/var/folders/sp/qr0jn46s10x5j0y77wb390d00000gn/T/ipykernel_1360/1741062523.py:2: FutureWarning: Series.__getitem__ treating keys as positions is deprecated. In a future version, integer keys will always be treated as labels (consistent with DataFrame behavior). To access a value by position, use `ser.iloc[pos]`
  if anova['PR(>F)'][0] < 0.05:


### Interpretação:
#### Valor F: 99.078295
#### Valor p (PR(>F)): 5.116294e-36

O valor p é extremamente pequeno (5.116294e-36), muito menor que o nível de significância comum (0.05). Isso indica que há uma diferença significativa nos salários entre os diferentes graus de instrução.


In [23]:
# Teste Post Hoc de Tukey para avaliar diferenças específicas
#  O teste de Tukey é realizado para identificar diferenças específicas entre os grupos de grau de instrução.x
tukey = pairwise_tukeyhsd(endog=df.salario, groups=df.grau_instrucao)
print("Resultados do Teste Post Hoc de Tukey:")
print(tukey.summary())

Resultados do Teste Post Hoc de Tukey:
Multiple Comparison of Means - Tukey HSD, FWER=0.05
group1 group2 meandiff p-adj lower  upper  reject
-------------------------------------------------
     f      m   3.2859   0.0 2.3414 4.2303   True
     f      s   6.5852   0.0 5.4835 7.6869   True
     m      s   3.2993   0.0 2.3071 4.2916   True
-------------------------------------------------


Interpretação Resultados do Teste Post Hoc de Tukey:
#### Colunas:

- **group1 e group2**: Representam os grupos de grau de instrução que estão sendo comparados.
- **meandiff**: Diferença de média entre os dois grupos comparados.
- **p-adj**: Valor p ajustado para múltiplas comparações. Um valor p muito pequeno (próximo de 0) indica que a diferença entre os grupos é estatisticamente significativa.
- **lower e upper**: Limites inferior e superior do intervalo de confiança para a diferença de média. Se o intervalo não inclui 0, a diferença é considerada significativa.
- **reject**: Indica se a hipótese nula (de que não há diferença entre as médias dos grupos) é rejeitada (True) ou não (False).

#### Por que os valores aparecem dessa forma:

- **Diferença de Média (meandiff)**: A diferença de média entre os grupos é calculada automaticamente pela função. Por exemplo, a diferença de média entre os grupos f e m é 3.2859.
- **Valor p Ajustado (p-adj)**: O valor p ajustado é muito pequeno (0.0), indicando que a diferença entre os grupos é altamente significativa.
- **Intervalo de Confiança (lower e upper)**: O intervalo de confiança para a diferença de média é calculado automaticamente. Por exemplo, para a comparação entre f e m, o intervalo de confiança é de 2.3414 a 4.2303.
- **Rejeição da Hipótese Nula (reject)**: A coluna reject é True para todas as comparações, indicando que a hipótese nula de que não há diferença entre as médias dos grupos é rejeitada para todas as comparações.

### Conclusão:

Os resultados do Teste Post Hoc de Tukey indicam que há diferenças significativas nos salários entre todos os pares de graus de instrução. Especificamente:

- O salário médio do grupo f é significativamente menor que o do grupo m.
- O salário médio do grupo f é significativamente menor que o do grupo s.
- O salário médio do grupo m é significativamente menor que o do grupo s.

Esses resultados sugerem que o grau de instrução tem um impacto significativo nos salários, com níveis mais altos de instrução associados a salários mais altos.
